src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -533,5 +539,141 @@ Json::Value Profile::ToJson() const
     JsonUtils::SetValueForKey(json, BellStyleKey, _BellStyle);
     JsonUtils::SetValueForKey(json, PixelShaderPathKey, _PixelShaderPath);
 
+    JsonUtils::SetValueForKey(json, EnvironmentKey, _EnvironmentVariables);
+
     return json;
 }
+
+namespace
+{
+    class VariablePair
+    {
+    public:
+        VariablePair(std::wstring rawValueIn) :
+            rawValue(std::move(rawValueIn))
+        {
+        }
+        VariablePair() = default;
+        virtual ~VariablePair() = default;
+        std::wstring rawValue;
+        std::wstring resolvedValue;
+    };
+
+    std::wstring ResolveEnvironmentVariableValue(const std::wstring& key, std::map<std::wstring, VariablePair>& envMap, std::list<std::wstring> seenKeys = {})

```

Comment:
My only _real_ concern with this PR is that this function needs more comments scattered throughout. It's not that easy to follow what exactly is going on, especially with the recursion.

Reply:
I've added some more comments, is it clear now or would you like more?

Reply:
Nah this is great, thanks!